src/html/mod.rs

```diff
@@ -1,9 +1,12 @@
-use crate::html::TokenType::{CommentToken, DoctypeToken, EndTagToken, ErrorToken, SelfClosingTagToken, StartTagToken, TextToken};
+use crate::html::TokenType::{
+    CommentToken, DoctypeToken, EndTagToken, ErrorToken, SelfClosingTagToken, StartTagToken,
+    TextToken,
+};
 use std::io::Read;
 use std::str;
 use std::string::ToString;
 
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
```

Comment:
If a type `T` derives `PartialEq` and all of its members implement `Eq`, then `T` can always implement `Eq`. Implementing `Eq` allows `T` to be used in APIs that require `Eq` types. It also allows structs containing `T` to derive `Eq` themselves.